File: core/src/main/java/com/lomekwi/cave/pipeline/num/NumFrame.java
package com.lomekwi.cave.pipeline.num;

import com.lomekwi.cave.pipeline.Frame;
import com.lomekwi.cave.pipeline.Source;
import com.lomekwi.cave.timeline.Track;

public class NumFrame extends Frame {
    private double val;
    public NumFrame(Track track) {
        super(track);
    }

    public NumFrame(Track track, Source<?> source) {
        super(track, source);
    }

    public double getVal() {
        return val;
    }

    public void setVal(double val) {
        this.val = val;
    }
}

File: core/src/main/scala/com/lomekwi/cave/pipeline/num/NumFrame.scala
package com.lomekwi.cave.pipeline.num

import com.lomekwi.cave.pipeline.Frame
import com.lomekwi.cave.pipeline.Source

class NumFrame(trackIndex: Int, source: Source[?]) extends Frame(trackIndex, source) {
  private var value: Double = 0

  def this(trackIndex: Int) = {
    this(trackIndex, null)
  }

  def getVal: Double = {
    value
  }

  def setVal(value: Double): Unit = {
    this.value = value
  }
}
